onpremise/onpremise.go

```diff
@@ -171,25 +178,104 @@ func WithCustomLogger(logger LogWriter) EngineOptions {
 
 		return nil
 	}
+}
+
+// ToggleFileWatch enables or disables file watching
+// in case 3rd party updates the data file on file system
+// engine will automatically reload the data file
+// default is true
+func ToggleFileWatch(enabled bool) EngineOptions {
+	return func(cfg *Engine) error {
+		cfg.isFileWatcherEnabled = enabled
+		return nil
+	}
+}
+
+// ToggleUpdateOnStart enables or disables update on start
+// default is false
+// if enabled, engine will pull the data file from the distributor on start of the engine
+func ToggleUpdateOnStart(enabled bool) EngineOptions {
```

Comment:
btw, can we name all EngineOptions consistently `With...`, like `WithUpdateOnStart`,  because some are `Set...`, some are `Toggle...` - it seems it would be more clear and consistent if we initialize the engine with all the options starting with `With...`